src/Simplex/Chat/Store/ContactRequest.hs

```diff
@@ -0,0 +1,299 @@
+                createBusinessRequestGroup db vr gVar user cReqChatVRange profile profileId ldn groupPreferences
+              liftIO $
+                DB.execute
+                  db
+                  "UPDATE contact_requests SET business_group_id = ? WHERE contact_request_id = ?"
+                  (groupId, contactRequestId)
+              ucr <- getContactRequest db user contactRequestId
+              pure $ RSCurrentRequest ucr (Just $ REBusinessChat gInfo clientMember) False
+    updateContactRequest :: UserContactRequest -> ExceptT StoreError IO RequestStage
+    updateContactRequest UserContactRequest {contactRequestId, contactId_, localDisplayName = oldLdn, profile = Profile {displayName = oldDisplayName}} = do
+      currentTs <- liftIO getCurrentTime
+      liftIO $ updateProfile currentTs
+      updateRequest currentTs
+      ucr' <- getContactRequest db user contactRequestId
+      re_ <- getRequestEntity ucr'
+      pure $ RSCurrentRequest ucr' re_ True
+      where
+        updateProfile currentTs =
+          DB.execute
+            db
+            [sql|
+              UPDATE contact_profiles
+              SET display_name = ?,
+                  full_name = ?,
+                  image = ?,
+                  contact_link = ?,
+                  updated_at = ?
+              WHERE contact_profile_id IN (
+                SELECT contact_profile_id
+                FROM contact_requests
+                WHERE user_id = ?
+                  AND contact_request_id = ?
+              )
+            |]
+            (displayName, fullName, image, contactLink, currentTs, userId, contactRequestId)
+        updateRequest currentTs =
+          if displayName == oldDisplayName
+            then
+              liftIO $
+                DB.execute
+                  db
+                  [sql|
+                    UPDATE contact_requests
+                    SET agent_invitation_id = ?, pq_support = ?, peer_chat_min_version = ?, peer_chat_max_version = ?, updated_at = ?
+                    WHERE user_id = ? AND contact_request_id = ?
+                  |]
+                  (Binary invId, pqSup, minV, maxV, currentTs, userId, contactRequestId)
+            else
+              ExceptT $ withLocalDisplayName db userId displayName $ \ldn -> runExceptT $ do
+                liftIO $ do
+                  DB.execute
+                    db
+                    [sql|
+                      UPDATE contact_requests
+                      SET agent_invitation_id = ?, pq_support = ?, peer_chat_min_version = ?, peer_chat_max_version = ?, local_display_name = ?, updated_at = ?
+                      WHERE user_id = ? AND contact_request_id = ?
+                    |]
+                    (Binary invId, pqSup, minV, maxV, ldn, currentTs, userId, contactRequestId)
+                  -- Here we could also update business chat, but is always synchronously auto-accepted so it's less of an issue
```

Comment:
this could be improved but currently ignored, as it's synchronously auto-accepted and profile update would be received via regular channel (unlike message change that wouldn't be sent as an update)